pkgs/development/libraries/sofia-sip/default.nix

```
@@ -20,6 +21,14 @@ stdenv.mkDerivation rec {
     sha256 = "sha256-7QmK2UxEO5lC0KBDWB3bwKTy0Nc7WrdTLjoQYzezoaY=";
   };
 
+  patches = [
+    # Fix build with gcc 14 from https://github.com/freeswitch/sofia-sip/pull/249
+    (fetchpatch {
+      url = "https://github.com/freeswitch/sofia-sip/pull/249/commits/46b02f0655af0a9594e805f09a8ee99278f84777.patch";
+      hash = "sha256-duvkBocn2otG84kC3E7V1ikl4uj4G+iB/dnAELzGFbI=";
+    })
```

You should use the commit URL instead of the PR's commit URL.

Also, GitHub's .patch can not always be reproduced. `.diff` can and is also smaller in size. See https://github.com/NixOS/nixpkgs/issues/257446

```suggestion
    (fetchpatch2 {
      name = "sofia-sip-fix-incomatible-pointer-type.patch";
      url = "https://github.com/freeswitch/sofia-sip/commit/46b02f0655af0a9594e805f09a8ee99278f84777.diff";
      hash = "sha256-4uZVtKnXG+BPW8byjd7tu4uEZo9SYq9EzTEvMwG0Bak=";
    })
```

---

Oops, yeah, ou're of course right.